### ISSUE TITLE
cv::VideoCapture memory cannot be manually released after reading video on different systems

### ISSUE THREAD

**Issue by hanjian-code** [opencv/opencv#26593]
### System Information

opencv 4.5.5

### Detailed description

The same program was run on ubuntu and windows, but the memory release of the program was inconsistent. my requirement is that the child thread releases the memory by itself after completing the task while the main thread continues to execute. However, I found that opencv cannot release the memory on Ubuntu, but it can on Windows. can you tell me why?

Uploading demo2.mp4…

Uploading WeChat_20241209163220.mp4…

### Steps to reproduce

#include <opencv2/opencv.hpp>
#include <thread>
#include <vector>
#include <iostream>

void video_capture_thread(int number)
{
	std::cout << "start -----------" << number << std::endl;
	
	std::string path = "E:\\workspace\\cpp\\test.mp4";

	cv::VideoCapture _capture(path);

	if (!_capture.isOpened())
	{
		std::cerr << path << std::endl;
		return;
	}

	cv::Mat _frame;
	int frame_count = 0;
	while (true)
	{
		_capture >> _frame;
		if (_frame.empty())
		{
			break;
		}
		if (frame_count % 100 == 0)
		{
			std::cout << "============frame_count============={}" << frame_count << std::endl;
		}
		frame_count += 1;
	}

	_frame.release();
	_capture.release();
	std::cout << "success -----------" << number << std::endl;
}

int main() {
	std::this_thread::sleep_for(std::chrono::seconds(10));
	std::vector<std::thread> _threads;
	int _number = 8;
	for (int i = 0; i < _number; ++i)
	{
		_threads.emplace_back(video_capture_thread, i);
	}

	for (auto& _thread : _threads)
	{
		_thread.detach();
	}

	std::this_thread::sleep_for(std::chrono::seconds(1000));

	return 0;
}

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)
